packages/grid/src/GridHOC.tsx

```diff
@@ -43,14 +97,306 @@ const GridHOC = ({ shouldFitContainer, ...rest }: IGrid) => {
+      ]);
+    } else {
+      onChangeColumnsFromProps([
+        ...leftMappedColumns,
+        ...centerMappedColumns,
+        ...cols,
+      ]);
+    }
+  }, [
+    leftMappedColumns,
+    centerMappedColumns,
+    rightMappedColumns,
+  ]);
+
+
+  if (isMultiGrid) {
+    const styles: CSSProperties = {
+      display: 'flex',
+    };
+
+    const multyGridProps = {
+      shouldMovingColumns,
+      shouldChangeColumnsWidth,
+      shouldChangeLeftColumnsWidth,
+      shouldChangeRightColumnsWidth,
+      width: 0,
+      height: 0,
+
+      leftFixedWidth,
+      rightFixedWidth,
+
+      wrapperSize,
+
+      leftMappedColumns,
+      centerMappedColumns,
+      rightMappedColumns,
+
+
+      setLeftMappedColumns,
+      setCenterMappedColumns,
+      setRightMappedColumns,
+
+      isScrollingOptOut,
+      overscanColumnCount,
+      overscanRowCount,
+      allGridsProps: {
+        onChangeColumns,
+        totals,
+        handleSelect,
+        onChangeExpand,
+        mappedItems,
+        selectedRows,
+        cacheRef,
+        themeRef: themeRef || {},
+        rowHeight,
+      },
+    };
+
+    if (shouldFitContainer) {
+      styles.height = '100%';
+      multyGridProps.width = wrapperSize.width;
+      multyGridProps.height = wrapperSize.height;
+
+      return (
+        <ScrollSync>
+          { ({
+            onScroll,
+            scrollTop,
+          }) => (
+            <div style={ styles } ref={ wrapperRef }>
+              <MultiGrid
+                { ...multyGridProps }
+                onScroll={ onScroll }
+                scrollTop={ scrollTop }
+              />
+            </div>
+          ) }
+        </ScrollSync>
+      );
+    }
+
+    multyGridProps.width = width as number;
+    multyGridProps.height = height as number;
+    return (
+      <ScrollSync>
+        { ({
+          onScroll,
+          scrollTop,
+        }) => (
+          <div style={ styles } ref={ wrapperRef }>
+            <MultiGrid
+              { ...multyGridProps }
+              onScroll={ onScroll }
+              scrollTop={ scrollTop }
+            />
+          </div>
+        ) }
+      </ScrollSync>
+    );
+  }
+
+
   if (shouldFitContainer) {
     return (
       <div ref={ wrapperRef } style={ { height: '100%' } }>
-        <Grid { ...rest } width={ wrapperSize.width } height={ wrapperSize.height } />
+        <Grid
+          handleSelect={ handleSelect }
+          onChangeExpand={ onChangeExpand }
+          mappedItems={ mappedItems }
+          selectedRows={ selectedRows }
+          width={ wrapperSize.width }
+          height={ wrapperSize.height }
+          cacheRef={ cacheRef }
+          themeRef={ themeRef }
+          rowHeight={ rowHeight }
+          shouldChangeColumnsWidth={ shouldChangeColumnsWidth }
+          gridHOCMappedColumns={ centerMappedColumns }
+          setGridHOCMappedColumns={ setCenterMappedColumns }
+          totals={ totals }
+          onChangeColumns={ onChangeColumns }
```

Comment:
I think. repeating props should move to variable and add this variable to each component